### PR TITLE
Mrge with bitbucket

### PR DESCRIPTION
**Pull request by Xiangyu-Hu**
Reverse StressRelaxationFirstHalf and include self-contact model

### CODE REVIEW

File: SPHINXsys/src/shared/particles/base_particles.cpp [hunk 1]
```diff
@@ -233,6 +233,14 @@ namespace SPH
 	void BaseParticles::writePltFileHeader(std::ofstream& output_file)
 	{
 		output_file << " VARIABLES = \"x\",\"y\",\"z\",\"ID\"";
+		
+		for (size_t l = 0; l != variables_to_write_[indexInteger].size(); ++l)
```
Comment (chuvirtonomy) [inline SPHINXsys/src/shared/particles/base_particles.cpp:237]: is it better to make this function less relying on class members? indexInteger/indexVector/variables_to_write_/all_particle_data_  are function input?

This way it is easier to write test for the function and code might be easier to manage.

**Review by chuvirtonomy — APPROVED** [review]

File: tests/2d_examples/test_2d_oscillating_beam/src/oscillating_beam.cpp [hunk 2]
```diff
@@ -75,82 +63,44 @@ std::vector<Vecd> CreatBeamShape()
...
 class Beam : public SolidBody
 {
 public:
 	Beam(SPHSystem &system, std::string body_name)
 		: SolidBody(system, body_name)
 	{
 		/** Geometry definition. */
-		std::vector<Vecd> beam_base_shape = CreatBeamBaseShape();
+		std::vector<Vecd> beam_base_shape = creatBeamBaseShape();
 		body_shape_ = new ComplexShape(body_name);
```
Comment (chuvirtonomy) [inline tests/2d_examples/test_2d_oscillating_beam/src/oscillating_beam.cpp:77]: when this is cleaned up?

**Review by chuvirtonomy — CHANGES_REQUESTED** [review]

**Review by chuvirtonomy — APPROVED** [review]
If resource cleanup is clear, no issue for me

**Review by BenceVirtonomy — APPROVED** [review]

File: SPHINXsys/src/shared/particles/base_particles.cpp (hunk 1 above)
Comment (Xiangyu-Hu) [reply SPHINXsys/src/shared/particles/base_particles.cpp:237]: Thanks for the comment ans I will update the suggested change in the next pull request.

File: tests/2d_examples/test_2d_oscillating_beam/src/oscillating_beam.cpp (hunk 2 above)
Comment (Xiangyu-Hu) [reply tests/2d_examples/test_2d_oscillating_beam/src/oscillating_beam.cpp:77]: The cleanup is slow but I will do it steady.

**Pull request merged**